src/tests/MetaDataQueryTests.cpp

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#include "KafkaW/MetaDataQueryImpl.h"
+#include "helpers/KafkaWMocks.h"
+#include <gtest/gtest.h>
+
+class CreateKafkaHandleTest : public ::testing::Test {};
+
+TEST_F(CreateKafkaHandleTest, Success) {
+  auto Result =
+      KafkaW::getKafkaHandle<RdKafka::Consumer, RdKafka::Conf>("some_broker");
+  EXPECT_FALSE(Result == nullptr);
+}
+
+class UsedProducerMock {
+public:
+  static int CallsToCreate;
+  static RdKafka::Producer *create(RdKafka::Conf *, std::string &) {
+    UsedProducerMock::CallsToCreate++;
+    return nullptr;
+  }
+};
+int UsedProducerMock::CallsToCreate = 0;
+
+TEST_F(CreateKafkaHandleTest, FailedToCreateHandle) {
+  UsedProducerMock::CallsToCreate = 0;
+  auto testFunc = [](auto Adr) { // Work around for GTest limitation
+    return KafkaW::getKafkaHandle<UsedProducerMock, RdKafka::Conf>(Adr);
+  };
+  EXPECT_THROW(testFunc("some_broker"), MetadataException);
+  EXPECT_EQ(UsedProducerMock::CallsToCreate, 1);
+}
+
+class UsedConfMock : public MockConf {
+public:
+  static int CallsToCreate;
+  static int CallsToSet;
+  static UsedConfMock *create(RdKafka::Conf::ConfType) {
+    UsedConfMock::CallsToCreate++;
+    return new UsedConfMock;
+  }
+  Conf::ConfResult set(std::string const &, std::string const &,
+                       std::string &) override {
+    UsedConfMock::CallsToSet++;
+    return RdKafka::Conf::CONF_INVALID;
+  }
+};
+int UsedConfMock::CallsToCreate = 0;
+int UsedConfMock::CallsToSet = 0;
+
+TEST_F(CreateKafkaHandleTest, FailedToSetBroker) {
+  UsedProducerMock::CallsToCreate = 0;
```

Comment:
This should be in the fixture if possible

Reply:
You should probably have a discussion with @matthew-d-jones about fixtures. 😃 Though I generally agree with you, in this particular case it would add code. Still, I have moved the lines anyway.